deeppavlov/models/kbqa/wiki_parser.py

```diff
@@ -253,32 +286,75 @@ def execute(self, what_return: List[str],
+                        elif fnd_num or value_str.isdigit():
+                            rel_comb[sort_elem] = float(value_str)
+                        new_rel_combs.append(rel_comb)
+                    new_rel_combs = [(elem, n) for n, elem in enumerate(new_rel_combs)]
+                    new_rel_combs = sorted(new_rel_combs, key=lambda x: (x[0][sort_elem], x[1]), reverse=reverse)
+                    new_rel_combs = [elem[0] for elem in new_rel_combs]
+                    new_rel_combs_list.append(new_rel_combs)
+                combs = [new_rel_combs[0] for new_rel_combs in new_rel_combs_list]
+
+            if what_return and what_return[-1].startswith("count"):
+                answers = [[len(combs)]]
             else:
-                combs = [[elem[key] for key in what_return] for elem in combs]
+                answers = [[elem[key] for key in what_return if key in elem] for elem in combs]
 
             if answer_types:
-                if answer_types == ["date"]:
-                    combs = [[entity for entity in comb
-                              if re.findall(r"[\d]{3,4}-[\d]{1,2}-[\d]{1,2}", entity)] for comb in combs]
+                if list(answer_types) == ["date"]:
+                    answers = [[entity for entity in answer
+                                if re.findall(r"[\d]{3,4}-[\d]{1,2}-[\d]{1,2}", entity)] for answer in answers]
+                elif list(answer_types) == ["not_date"]:
+                    answers = [[entity for entity in answer
+                                if not re.findall(r"[\d]{3,4}-[\d]{1,2}-[\d]{1,2}", entity)] for answer in answers]
                 else:
                     answer_types = set(answer_types)
-                    combs = [[entity for entity in comb
-                              if answer_types.intersection(self.find_types(entity))] for comb in combs]
-                combs = [comb for comb in combs if any([entity for entity in comb])]
-
-        return combs
+                    answers = [[entity for entity in answer
+                                if answer_types.intersection(self.find_types(entity))] for answer in answers]
+            if is_boolean:
+                answers = [["Yes" if len(triplets) > 0 else "No"]]
+            found_rels = [[elem[key] for key in rels_from_query if key in elem] for elem in combs]
+            ans_rels_combs = [(answer, rel, comb) for answer, rel, comb in zip(answers, found_rels, combs)
+                                if any([entity for entity in answer])]
+            answers = [elem[0] for elem in ans_rels_combs]
+            found_rels = [elem[1] for elem in ans_rels_combs]
+            found_combs = [elem[2] for elem in ans_rels_combs]
+
+        return answers, found_rels, found_combs
+
+    def define_is_boolean(self, query_hdt_seq):
+        is_boolean = False
+        if len(query_hdt_seq) == 1 and all([not query_hdt_seq[0][i].startswith("?") for i in [0, 2]]):
+            is_boolean = True
+        return is_boolean
```

Comment:
whole method body could be replaced with
```
return len(query_hdt_seq) == 1 and all([not query_hdt_seq[0][i].startswith("?") for i in [0, 2]])
```
Also, method is used only one time, so it could be removed and if construction could be placed in the `execute` method.